sky/engine/core/script/dart_init.cc

```diff
@@ -146,7 +146,9 @@ Dart_Isolate IsolateCreateCallback(const char* script_uri,
       const SkySettings& settings = SkySettings::Get();
       if (settings.enable_observatory) {
         std::string ip = "127.0.0.1";
-        const intptr_t port = settings.observatory_port;
+        // We auto select the port on Android and iOS simulators.
+        const intptr_t port =
+            IsRunningPrecompiledCode() ? settings.observatory_port : 0;
```

Comment:
Rather than ignoring the setting, maybe we should make the default value of the setting zero (i.e., in sky_settings.h).